core/src/engine/ev/thread.cpp

```diff
@@ -147,14 +160,24 @@ void Thread::Start(const std::string& name) {
   ev_set_priority(&watch_break_, EV_MAXPRI);
   ev_async_start(loop_, &watch_break_);
 
+  using LibEvDuration = std::chrono::duration<double>;
   if (register_event_mode_ == RegisterEventMode::kDeferred) {
-    using LibEvDuration = std::chrono::duration<double>;
     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
     ev_timer_init(
         &timers_driver_, UpdateTimersWatcher, 0.0,
         std::chrono::duration_cast<LibEvDuration>(kPeriodicEventsDriverInterval)
             .count());
     ev_timer_start(loop_, &timers_driver_);
+  } else {
+    // We set up a noop high-interval timer here to wake up the loop: that
+    // helps to avoid cpu-stats stall with outdated values if we become idle
+    // after a burst. No point in doing so if we already have timers_driver.
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast)
+    ev_timer_init(
+        &stats_timer_, UpdateTimersWatcher, 0.0,
+        std::chrono::duration_cast<LibEvDuration>(kCpuStatsCollectInterval)
+            .count());
```

Comment:
It adds some visible cpu overhead, doesn't it?

Reply:
not at all - it's one wakeup / s, with 200 ev threads CPU usage is about 0.7 - 1.3%.